yaspin/core.py

```diff
@@ -357,14 +350,19 @@ def _spin(self):
 
             # Write
             with self._stdout_lock:
-                sys.stdout.write(out)
                 self._clear_line()
+                sys.stdout.write(out)
                 sys.stdout.flush()
+                self._cur_line_len = max(self._cur_line_len, len(out))
 
             # Wait
             self._stop_spin.wait(self._interval)
 
     def _compose_color_func(self):
+        if not sys.stdout.isatty():
+            # ANSI Color Control Sequences are problematic in Jupyter
+            return None
```

Comment:
IMO if the purpose of this code block is to disable colors usage in Jupyter, it should be done during spinner instantiation by throwing e.g. `ValueError`. Also, there might be a library which provides colors support for Jupyter.

Reply:
Exactly, it disables color usage.

Maybe there is a such library, but to be honest, I don't need color support right now. Maybe this is something for a future PR? The issue with throwing a `ValueError` on init is that it might break existing code: now, it "works" in Jupyter, but just prints some ugly characters. Then, the entire cell execution would fail with a `ValueError`.

I'd rather suggest using the `warnings` module, so that when you pass a color during init, you just get a warning printed (that can be silenced).

Reply:
Agree, let's handle colors support for Jupyter in future. Also agree that throwing exceptions during spinner instantiation is a bad idea.